File: Chatbot-Studi.com/PublicSiteChatbot.Front/Studi.AI.Chatbot.Front/Client/UserRequestModel.cs
namespace Studi.AI.Chatbot.Front.Client;

using Studi.AI.Chatbot.Front.Models;
using System.Text.Json.Serialization;

public class UserRequestModel
{
    [JsonPropertyName("user_id")]
    public Guid? UserId { get; set; }

    [JsonPropertyName("user_name")]
    public string UserName { get; set; } = "";

    [JsonPropertyName("IP")]
    public string IP { get; set; } = "";

    [JsonPropertyName("device_info")]
    public DeviceInfoRequestModel DeviceInfo { get; set; } = new();
}


public class DeviceInfoRequestModel
{
    [JsonPropertyName("user_agent")]
    public string UserAgent { get; set; } = string.Empty;

    [JsonPropertyName("platform")]
    public string Platform { get; set; } = string.Empty;

    [JsonPropertyName("app_version")]
    public string AppVersion { get; set; } = string.Empty;

    [JsonPropertyName("os")]
    public string Os { get; set; } = string.Empty;

    [JsonPropertyName("browser")]
    public string Browser { get; set; } = string.Empty;

    [JsonPropertyName("is_mobile")]
    public bool IsMobile { get; set; } = false;

    public static DeviceInfoRequestModel FromModel(DeviceInfoModel deviceInfoModel)
    {
        return new DeviceInfoRequestModel
        {
            UserAgent = deviceInfoModel.UserAgent,
            Platform = deviceInfoModel.Platform,
            AppVersion = deviceInfoModel.AppVersion,
            Os = deviceInfoModel.Os,
            Browser = deviceInfoModel.Browser,
            IsMobile = deviceInfoModel.IsMobile
        };
    }
}

File: Chatbot-Studi.com/Chatbot.BackOffice.Front/client_models/user_request_model.py
from typing import Optional
from uuid import UUID

from pydantic import BaseModel

class DeviceInfoRequestModel(BaseModel):
    user_agent: str = ""
    platform: str = ""
    app_version: str = ""
    os: str = ""
    browser: str = ""
    is_mobile: bool = False

    def to_json(self) -> dict:
        return {
            "user_agent": self.user_agent,
            "platform": self.platform,
            "app_version": self.app_version,
            "os": self.os,
            "browser": self.browser,
            "is_mobile": self.is_mobile,
        }
    
class UserRequestModel(BaseModel):
    user_id: Optional[UUID] = None
    user_name: str = ""
    IP: str = ""
    device_info: Optional[DeviceInfoRequestModel] = DeviceInfoRequestModel()

    def to_json(self) -> dict:
        return {
            "user_id": str(self.user_id) if self.user_id else None,
            "user_name": self.user_name,
            "IP": self.IP,
            "device_info": self.device_info.to_json(),
        }